packages/atclient/src/atclient.c

```diff
@@ -239,3 +243,300 @@ void atclient_free(atclient *ctx) {
   atclient_connection_free(&(ctx->root_connection));
   atclient_connection_free(&(ctx->secondary_connection));
 }
+
+static int atclient_create_shared_encryption_key(atclient *ctx, const atclient_atsign *recipient,
+                                                 char *enc_key_shared_by_me);
+
+static int atclient_get_public_encryption_key(atclient *ctx, const atclient_atsign *atsign,
+                                              char *public_encryption_key);
+
+int atclient_get_encryption_key_shared_by_me(atclient *ctx, const atclient_atsign *recipient,
+                                             char *enc_key_shared_by_me) {
+  int ret = 1;
+
+  // llookup:shared_key.recipient_atsign@myatsign
+  char *command_prefix = "llookup:shared_key.";
+  short command_prefix_len = 19;
+  short atsign_with_at_len = (short)strlen(ctx->atsign.atsign);
+
+  short command_len = command_prefix_len + (atsign_with_at_len * 2 - 1) + 3;
+  char command[command_len];
+  snprintf(command, command_len, "llookup:shared_key.%s%s\r\n", recipient->without_prefix_str, ctx->atsign.atsign);
+
+  const unsigned long recvlen = 1024;
+  unsigned char recv[sizeof(unsigned char) * recvlen];
+  memset(recv, 0, sizeof(unsigned char) * recvlen);
```

Comment:
Here, we do need to multiply by `sizeof(unsigned char)` so this line is correct.